io/disk/htx_block_devices.py

```diff
@@ -68,38 +72,86 @@ def setup_htx(self):
         Builds HTX
         """
         packages = ['git', 'gcc', 'make']
-        detected_distro = distro.detect().name
-        if detected_distro in ['centos', 'fedora', 'rhel', 'redhat']:
+        self.detected_distro = distro.detect()
+        if self.detected_distro.name in ['centos', 'fedora', 'rhel', 'redhat']:
             packages.extend(['gcc-c++', 'ncurses-devel', 'tar'])
-        elif detected_distro == "Ubuntu":
+        elif self.detected_distro.name == "Ubuntu":
             packages.extend(['libncurses5', 'g++', 'ncurses-dev',
                              'libncurses-dev', 'tar'])
-        elif detected_distro == 'SuSE':
+        elif self.detected_distro.name == 'SuSE':
             packages.extend(['libncurses5', 'gcc-c++', 'ncurses-devel', 'tar'])
         else:
-            self.cancel("Test not supported in  %s" % detected_distro)
+            self.cancel("Test not supported in  %s" % self.detected_distro)
 
         smm = SoftwareManager()
         for pkg in packages:
             if not smm.check_installed(pkg) and not smm.install(pkg):
                 self.cancel("Can not install %s" % pkg)
 
-        url = "https://github.com/open-power/HTX/archive/master.zip"
-        tarball = self.fetch_asset("htx.zip", locations=[url], expire='7d')
-        archive.extract(tarball, self.teststmpdir)
-        htx_path = os.path.join(self.teststmpdir, "HTX-master")
-        os.chdir(htx_path)
-
-        exercisers = ["hxecapi_afu_dir", "hxedapl", "hxecapi", "hxeocapi"]
-        for exerciser in exercisers:
-            process.run("sed -i 's/%s//g' %s/bin/Makefile" % (exerciser,
-                                                              htx_path))
-        build.make(htx_path, extra_args='all')
-        build.make(htx_path, extra_args='tar')
-        process.run('tar --touch -xvzf htx_package.tar.gz')
-        os.chdir('htx_package')
-        if process.system('./installer.sh -f'):
-            self.fail("Installation of htx fails:please refer job.log")
+        if self.run_type == 'git':
+            url = "https://github.com/open-power/HTX/archive/master.zip"
+            tarball = self.fetch_asset("htx.zip", locations=[url], expire='7d')
+            archive.extract(tarball, self.teststmpdir)
+            htx_path = os.path.join(self.teststmpdir, "HTX-master")
+            os.chdir(htx_path)
+
+            exercisers = ["hxecapi_afu_dir", "hxedapl", "hxecapi", "hxeocapi"]
```

Comment:
can we have exercisers from yaml param.. so we can always change

Reply:
@abdhaleegit This is something to do internally and not an input here.
So I think having this internally is ok I feel.